lib/client-options.js

```diff
@@ -44,7 +44,8 @@ function defaultOptions () {
     authProvider: null,
     maxPrepared: 500,
     encoding: {
-      copyBuffer: true
+      copyBuffer: true,
+      useUndefinedAsUnset: true
```

Comment:
This will be a problem for those using < 2.2 right?  Since we have the protocolVersion what do you think about adding a check to encode in that we only set a value to unset if protocolVersion >= 4?